neural-style-util.js

```diff
@@ -123,35 +124,12 @@ exports.saveImage = function(id, purpose, data, callback) {
 }
 
 exports.findImagePath = function(id, purpose, callback) {
-  var pathPrefix = exports.getImagePathPrefix(id, purpose);
-  function makeExtensionCheck(ext) {
-    return function(cb) {
-      var path = pathPrefix + '.' + ext;
-      fs.stat(path, function(err, stats) {
-        if (!err) {
-          cb(null, path);
-        } else {
-          cb(null, null);
-        }
-      });
-    };
-  }
-  async.parallel([
-    makeExtensionCheck('jpg'),
-    makeExtensionCheck('png'),
-  ], function(err, results) {
-    if (err) {
-      callback(err);
-      return;
-    }
-    var result = _.find(results, function(result) {
-      return result != null;
-    });
-    if (result) {
-      callback(null, result);
-    } else {
-      callback(new Error('Missing image with path prefix ' + pathPrefix));
-    }
+  var fileNamePrefix = id + '_' + purpose;
+  exec("find "+config.get('dataPath')+" -name '"+fileNamePrefix+"*'", function(error, stdout, stderr){
```

Comment:
Instead of using exec to run find (which could maybe be different on different systems) I'd prefer to use build-in Node fs functions for the file matching (e.g. fs.readdir, and then globbing or regex matching on the resulting file list).